### DISCUSSION TITLE
HM-MOD-RPI-PCB in Docker-Variante nicht gefunden

### DISCUSSION THREAD

**Discussion by winnepar** [OpenCCU/OpenCCU#2191] [🇩🇪 Generelle Diskussionen]
Hi, I used a lot of years RaspberryMatic as stand-alone on Raspberry (with Hm- and HmIP-modules). Now I want to use RaspberryMatic in a docker, I followed the installation manual DICKER/OCI (includes Point 5 and docker-compose.yml) of Jens, but the RF-Module isn't found anytime. What can i do)

**Comment by jens-maus** [comment]
Sorry, but this is way too less information. No logs, no bootup messages, nothing. Just your statement "Doesn't work!". How do you think your problem should be analyzed or investigated with the provided information?!?!

**Comment by winnepar** [comment]
danke für die schnelle Antwort. (Ich bin mir nicht sicher , ob ENG die richtige Sprachwahl ist --> deshalb jetzt mal in DEU: geht bei mir schneller und präziser). Nach absenden ds Problems war ich mir schon unsicher, ob die Infos reichen, hier alles, was mir einfällt:

HW:   Raspberry 4 / 2 GB (neu aufgesetzt)

         wahlweise Funkmodul HM-MOD-RPI-PCM bzw. RPI-RF-MOD

OS:    PI OS 64Bit  mit Desktob-Erweiterung (für VNC-Zugriff)

          VNC + SSH freigeschaltet + getestet

          arbeite prinzipiell als root (s.u.)

SW:    Docker version 20.10.23, build 7155243        

          docker-compose version 1.29.2, build unknown

         
Container:
 ```
***@***.***:/# docker ps
CONTAINER ID   IMAGE                                     COMMAND                  CREATED        STATUS                  PORTS                                                                                                                                                                                                                                                                                                                                                                                                                   NAMES
5a59205257f6   ghcr.io/jens-maus/raspberrymatic:latest   "/sbin/init"             18 hours ago   Up 12 hours (healthy)   22/tcp, 161/tcp, 443/tcp, 1900-1902/tcp, 0.0.0.0:2001->2001/tcp, :::2001->2001/tcp, 1999-2000/tcp, 5987/tcp, 0.0.0.0:2010->2010/tcp, :::2010->2010/tcp, 8088/tcp, 0.0.0.0:8181->8181/tcp, :::8181->8181/tcp, 9099/tcp, 9293/tcp, 10000/tcp, 41999-42001/tcp, 42010/tcp, 43438-43439/tcp, 48181/tcp, 48899/tcp, 49292/tcp, 49880/tcp, 0.0.0.0:9292->9292/tcp, :::9292->9292/tcp, 0.0.0.0:8080->80/tcp, :::8080->80/tcp   5a59205257f6_ccu
56bb585be98c   buanet/iobroker                           "/bin/bash -c /opt/s…"   21 hours ago   Up 12 hours (healthy)   0.0.0.0:1880->1880/tcp, :::1880->1880/tcp, 0.0.0.0:1883->1883/tcp, :::1883->1883/tcp, 0.0.0.0:8091->8081/tcp, :::8091->8081/tcp                                                                                                                                                                                                                                                                                         iobroker
***@***.***:/#
```
 
Vorgehen lt. Deiner WiKi-Seite Installation Docker OCI  ... bis

3. OK

4.


...

rpi_rf_mod_led.ko:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/5.10.0-21-arm64/updates/dkms/

rtc-rx8130.ko:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/5.10.0-21-arm64/updates/dkms/

depmod...

DKMS: install completed.
.
linux-headers-arm64 (5.10.162-1) wird eingerichtet ...
Trigger für man-db (2.9.4-2) werden verarbeitet ...
***@***.***:/opt/containers/raspmatic#


--> OK.

5.
...

depmod....

DKMS: install completed.
.
pivccu-modules-raspberrypi (2.0-17) wird eingerichtet ...
»Umleitung von /var/lib/piVCCU/dtb/overlays/pivccu-raspberrypi.dtbo zu /usr/share/rpikernelhack/overlays/pivccu
-raspberrypi.dtbo durch rpikernelhack« wird entfernt
Created symlink /etc/systemd/system/multi-user.target.wants/pivccu-rpi-modules.service → /lib/systemd/system/pi
vccu-rpi-modules.service.
***@***.***:/opt/containers/raspmatic#


...

8. --> OK

 

Raspberrymatic mit docker-compose.yml installiert.

 

In Anhang (docker logs 5a59205257f6_ccu > Rasp.txt) ab Zeile 16: ... RF-Hardware ... n/a.

 

Einrichten/Start über <ip>:8080 funktioniert, aber im Anlern-Modus "... keine RF-Modl ..." ... :-(

 

Was verrückt ist: Die Hardware funktioniert, wenn ich RaspberryMatic direkt installiere und starte, funktionieren beide Funk-Module und das Anlernen!

 

Entschuldige meine Generve, aber ich (67) weiß wirklich nicht mehr weiter.

**Reply by jens-maus** [reply]
BTW: Bitte nicht per Email auf GitHub Benachrichtigungen antworten. Man nutzt dafür das Webinterface von GitHub, sonst kommt hier nur unformatiertes durcheinander rein wie man an deinem letzten Reply sieht.

Bzgl. deines Problems würde ich vorschlagen – wenn du mit Docker wenig bis keine Erfahrung hast – davon abstand zu nehmen und RaspberryMatic auf einem Pi direkt zu installieren und nicht zu versuchen eine Eierlegene Wollmichsau zu bauen in dem du alles auf eine Karte/Pi setzt. Das aufsetzen eines Docker-basierten RaspberryMatic kann so seine Tücken haben und wenn man in solche Probleme wie du reinrennst dann ist es mitunter schwer dort aus der ferne zu helfen, etc. Da muss man dann schon wissen was man tut bzw. tun muss um die Probleme selbst zu lösen. Wie in den meisten bisherigen Fällen wird es auch hier sein das es an einer eigenheit des Host-OS liegt da hier ggf. das Homematic Funkmodul nicht gefunden bzw. nicht richtig initialisiert wird. Und auch das kann man nur mit direktem Zugriff bzw. mehr detail infos hier lösen. Du zeigst ja auch nicht mal die Startup-Meldungen des Containers bzw. den ausgaben von `/var/log/boot.log` das man am schluss innerhalb des Containers findet.

Also bitte einfach mal überlegen ob ein dockerisiertes RaspberryMatic wirklich das richtige für dich ist und ob du nicht besser auf ein vollwertiges bzw. vollvirtualisiertes (Proxmox&Co) RaspberryMatic setzen willst. Denn auch wenn wir das Problem jetzt gelöst bekommen, beim nächsten RaspberryPiOS Update (z.B. Kernel Update) wirst du sicher wieder vor dem selben Problem stehen.

**Comment by winnepar** [comment]
dnake für Deine Hinweise/Tipps. Werde das ganze "Konzept" nochmal überdenken ...
